client/gribiclient.go

```diff
@@ -792,17 +806,32 @@ func (c *Client) clearPendingOp(op *spb.AFTResult) (*OpResult, error) {
 	c.qs.pendMu.Lock()
 	defer c.qs.pendMu.Unlock()
 	v := c.qs.pendq.Ops[op.Id]
-	if v == nil {
-		return nil, fmt.Errorf("could not dequeue operation %d, unknown operation", op.Id)
-	}
 	switch op.GetStatus() {
 	case spb.AFTResult_FIB_PROGRAMMED, spb.AFTResult_FIB_FAILED:
-		delete(c.qs.pendq.Ops, op.Id)
+		switch {
+		case TreatRIBACKAsCompletedInFIBACKMode:
```

Comment:
You're right -- apologies, I'd forgotten that `RIB_ACK` is optional in `RIB_AND_FIB_ACK` mode. I'll actually refactor this to say:

```
switch TreatRIBACKAsCompletedInFIBACKMode {
case true:
   // handle RIB_ACK - complete transaction and dequeue
   // handle FIB_ACK - if transaction still exists, dequeue - log that there was no RIB ACK
   // handle failed case
case false:
   // previous handling
}
```

I think this will make things clearer. In the case that a user sets the `TreatRIBACK...` flag and we never see a `RIB_ACK` then they are doing something strange, we're adding this specifically to be able to benchmark a gRIBI implementation's `RIB_ACK` response time even though we're asking for `FIB_ACK` (I'll share some data as to why with you out-of-band of this PR) -- and in the case that the server doesn't send `RIB_ACK` then it's not clear what the user is thinking.

Production-wise, per the comment, this flag should never be true -- `RIB_AND_FIB_ACK` means that an operation is ONLY terminated when a failed, or `FIB_ACK` is received AFAICS. `AwaitConverged` shouldn't return and say this was OK (recalling that if the server can timeout the operation).

WDYT?

Reply:
yes, I think this looks even better.

Reply:
Implemented, much cleaner. PTAL!